src/parallel.ts

```diff
@@ -1,29 +1,10 @@
 import assert from "assert";
 import deepExtend from "deep-extend";
 
-export function variablesList (jobData: any, jobName: string) {
-    if (jobData?.parallel !== null) {
-        if (Array.isArray(jobData.parallel?.matrix)) {
-            const matrixVariables = matrixVariablesList(jobData, jobName) ?? [null];
-            return {
-                matrixVariables: matrixVariables,
-                nodesTotal: matrixVariables?.length,
-            };
-        } else if (Number.isInteger(jobData.parallel)) {
-            return {
-                matrixVariables: Array(jobData.parallel).fill(null),
-                nodesTotal: jobData.parallel,
-            };
-        }
-    }
-
-    return {
-        matrixVariables: [null],
-        nodesTotal: null,
-    };
-}
-
 export function matrixVariablesList (jobData: any, jobName: string) {
+    if (Number.isInteger(jobData.parallel)) {
+        return Array(jobData.parallel).fill(null);
+    }
```

Comment:
I have implemented this simplified way like you suggested. Tell me which version you prefer and I will either drop this commit or squash it with the previous one.

Reply:
personally, i'll prefer this